Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -578,6 +578,24 @@ theorem pow_mem_of_mem (ha : a ∈ I) (n : ℕ) (hn : 0 < n) : a ^ n ∈ I :=
     (fun m _hm => (pow_succ a m).symm ▸ I.mul_mem_right (a ^ m) ha) hn
 #align ideal.pow_mem_of_mem Ideal.pow_mem_of_mem
 
+theorem pow_mem_of_pow_mem {m n : ℕ} (ha : a ^ m ∈ I) (h : m ≤ n) : a ^ n ∈ I := by
+  rw [← Nat.add_sub_of_le h, pow_add]
+  exact I.mul_mem_right _ ha
+
+theorem add_pow_add_pred_mem_of_pow_mem  {m n : ℕ}
+    (ha : a ^ m ∈ I) (hb : b ^ n ∈ I) :
+    (a + b) ^ (m + n - 1) ∈ I := by
+  rw [add_pow]
+  apply I.sum_mem
+  intro c _
+  apply mul_mem_right
+  by_cases h : m ≤ c
+  · exact I.mul_mem_right _ (I.pow_mem_of_pow_mem ha h)
+  · refine I.mul_mem_left _ (I.pow_mem_of_pow_mem hb ?_)
+    simp only [not_le, Nat.lt_iff_add_one_le] at h
+    rw [Nat.sub_sub, add_comm 1, add_comm m, Nat.add_sub_assoc h]
+    apply Nat.le_add_right
+
```

Comment:
How about:
```suggestion
variable {I}

theorem add_pow_mem_of_pow_mem_of_le {m n k : ℕ}
    (ha : a ^ m ∈ I) (hb : b ^ n ∈ I) (hk : m + n ≤ k + 1) :
    (a + b) ^ k ∈ I := by
  rw [(Commute.all _ _).add_pow']
  refine I.sum_mem fun ⟨i, j⟩ hij ↦ ?_
  rw [nsmul_eq_mul]
  apply mul_mem_left
  by_cases h : m ≤ i
  · exact I.mul_mem_right _ (I.pow_mem_of_pow_mem ha h)
  · refine I.mul_mem_left _ (I.pow_mem_of_pow_mem hb ?_)
    linarith [Finset.mem_antidiagonal.mp hij]

theorem add_pow_add_pred_mem_of_pow_mem  {m n : ℕ}
    (ha : a ^ m ∈ I) (hb : b ^ n ∈ I) :
    (a + b) ^ (m + n - 1) ∈ I :=
  add_pow_mem_of_pow_mem_of_le ha hb <| by rw [← Nat.sub_le_iff_le_add]

```

Reply:
I'll do that, as well as for the nilpotent version.